.github/workflows/ci-fusilli-plugin.yml

```diff
@@ -0,0 +1,100 @@
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: CI - fusilli-plugin
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+concurrency:
+  # A PR number if a pull request and otherwise the commit hash. This cancels
+  # queued and in-progress runs for the same PR (presubmit) or commit
+  # (postsubmit). The workflow name is prepended to avoid conflicts between
+  # different workflows.
+  group: ${{ github.workflow }}-${{ github.event.number || github.sha }}
+  cancel-in-progress: true
+
+env:
+  FUSILLI_DIR: ${{ github.workspace }}/sharkfuser
+  FUSILLI_PLUGIN_DIR: ${{ github.workspace }}/fusilli-plugin
+
+jobs:
+  build-and-test:
+    name: "Build & Test :: ${{ matrix.name }}"
+    runs-on: ${{ matrix.runs-on }}
+    defaults:
+      run:
+        # --noprofile skips loading ~/.bash_profile
+        # --norc      skips loading ~/.bashrc
+        # -exo pipefail:
+        #   -e exit immediately if any command fails
+        #   -x print each command before executing
+        #   -o pipefail ensure pipeline fail sif any command in it fails
+        # {0} github actions placeholder for the command to run
+        shell: bash --noprofile --norc -exo pipefail {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: ["gfx942_clang18_debug"]
+        include:
+          - name: gfx942_clang18_debug
+            runs-on: linux-mi325-2gpu-ossci-nod-ai
+            fusilli-plugin-cmake-options:
+              -DCMAKE_C_COMPILER=clang-18
+              -DCMAKE_CXX_COMPILER=clang++-18
+              -DCMAKE_LINKER_TYPE=LLD
+              -DFUSILLI_SYSTEMS_AMDGPU=ON
+
+    steps:
+    - name: checkout fusilli-plugin
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    # Pass FUSILLI_CACHE_DIR=/tmp to configure the compilation cache
+    # to be written to /tmp. It defaults to $HOME when not set but there
+    # are permissions issues with GitHub runners when accessing $HOME.
+    # TODO: remove when fix is found
+    - name: build fusilli-plugin
+      # The docker container mounts the working directory as a volume, so we
+      # must run from github.workspace to ensure fusilli-plugin can reach into
+      # ../sharkfuser for its fusilli dependency.
+      working-directory: ${{ github.workspace }}
+      run: |
+        ${{ env.FUSILLI_DIR }}/build_tools/docker/exec_docker_ci.sh \
+        bash -c "export FUSILLI_CACHE_DIR=/tmp && \
+                 cd fusilli-plugin && \
+                 cmake -GNinja -S. -Bbuild \
+                   -DIREERuntime_DIR=/workspace/.cache/docker/iree/build/lib/cmake/IREE \
+                   ${{ matrix.fusilli-plugin-cmake-options }} && \
+                 cmake --build build --target all &&
+                 ctest --test-dir build --output-on-failure --extra-verbose --timeout 30"
```

Comment:
Now I know where it was picking the timeout of 30 seconds for Fusilli! When it was failing due to timeout in CI, I instead increased the TIMEOUT in `FusilliTestUtils.cmake` to 60 seconds.